synapse/storage/databases/main/events_worker.py

```diff
@@ -62,7 +62,8 @@
 EVENT_QUEUE_THREADS = 3  # Max number of threads that will fetch events
 EVENT_QUEUE_ITERATIONS = 3  # No. times we block waiting for requests for events
 EVENT_QUEUE_TIMEOUT_S = 0.1  # Timeout when waiting for requests for events
-
+EVENT_EXPIRY_TIME = 24 * 60 * 60 * 1000
```

Comment:
Not sure if we really want to have a timeout at all. 24h seemed reasonable at the time, but typically this means the cache gets thrashed every 24 hours. Maybe we should have no limit and expect admins to regulate redis memory usage through configuration?

Reply:
I removed this, there was a significant performance penalty when the events eventually expired out of the cache. Better to let redis handle cache eviction.

Reply:
An alternative would be to update the expiry on each access

Reply:
Done!